From eac7d056d7a0cc9c137a6b19297d103caa8c43d8 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Fri, 29 Aug 2014 21:18:40 -0400
Subject: [net] svcrpc: fix error-handling on badd gssproxy downcall

Message-id: <1409347125-11495-2-git-send-email-bfields@redhat.com>
Patchwork-id: 89913
O-Subject: [Patch RHEL7 1/6] svcrpc: fix error-handling on badd gssproxy downcall
Bugzilla: 1117914
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

For every other problem here we bail out with an error, but here for
some reason we're setting a negative cache entry (with, note, an
undefined expiry).

It seems simplest just to bail out in the same way as we do in other
cases.

Cc: Simo Sorce <simo@redhat.com>
Reported-by: Andi Kleen <andi@firstfloor.org>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
(cherry picked from commit 3be34555fa294483202483c4b171a8d335447187)
Bugzilla 1117914

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/sunrpc/auth_gss/svcauth_gss.c b/net/sunrpc/auth_gss/svcauth_gss.c
index 32afef9..0ce7c26 100644
--- a/net/sunrpc/auth_gss/svcauth_gss.c
+++ b/net/sunrpc/auth_gss/svcauth_gss.c
@@ -1167,8 +1167,8 @@ static int gss_proxy_save_rsc(struct cache_detail *cd,
  if (!ud->found_creds) {
   /* userspace seem buggy, we should always get at least a
    * mapping to nobody */
-  dprintk("RPC:       No creds found, marking Negative!\n");
-  set_bit(CACHE_NEGATIVE, &rsci.h.flags);
+  dprintk("RPC:       No creds found!\n");
+  goto out;
  } else {
 
   /* steal creds */
-- 
1.7.1